src_tests/test_uniform_losses_boson_sampling_simulator.py

```diff
@@ -0,0 +1,109 @@
+        self.initial_state = [1, 1, 1, 1, 0]
+
+        # Create configuration object.
+        self.experiment_configuration = BosonSamplingExperimentConfiguration(
+            interferometer_matrix=self.permutation_matrix,
+            initial_state=self.initial_state,
+            initial_number_of_particles=sum(self.initial_state),
+            number_of_modes=len(self.initial_state),
+            number_of_particles_lost=self.number_of_particles_lost,
+            number_of_particles_left=sum(self.initial_state) - self.number_of_particles_lost,
+            probability_of_uniform_loss=0.8
+        )
+
+        self.simulation_strategy = \
+            UniformLossSimulationStrategy(self.permutation_matrix,
+                                          self.experiment_configuration.number_of_modes,
+                                          self.experiment_configuration.probability_of_uniform_loss)
+
+        self.simulator = BosonSamplingSimulator(self.experiment_configuration.number_of_particles_left,
+                                                self.experiment_configuration.initial_number_of_particles,
+                                                self.experiment_configuration.number_of_modes,
+                                                self.simulation_strategy)
+
+    def test_approximate_and_exact_distribution_distance(self) -> None:
+        exact_distribution_calculator = \
+            BosonSamplingWithUniformLossesExactDistributionCalculator(self.experiment_configuration)
+        exact_distribution = exact_distribution_calculator.calculate_exact_distribution()
+        # TR TODO: Delta in assertion should depend on samples number!
+        approximate_distribution = self.__calculate_approximate_distribution()
+        distance = calculate_total_variation_distance(exact_distribution, approximate_distribution)
+        bound = self.__calculate_uniform_loss_distribution_error_bound()
+        self.assertAlmostEqual(distance, bound, delta=1e-2)
+
+    def __calculate_approximate_distribution(self, samples_number: int = 10000) -> List[float]:
+        """
+        Prepares the approximate distribution using boson sampling simulation method described by
+        Oszmaniec and Brod. Obviously higher number of samples will generate better approximation.
+        :return: Approximate distribution as a list.
+        """
+        exact_distribution_calculator = \
+            BosonSamplingWithUniformLossesExactDistributionCalculator(self.experiment_configuration)
+
+        possible_outcomes = exact_distribution_calculator.get_outcomes_in_proper_order()
+
+        outcomes_probabilities = zeros(len(possible_outcomes))
+
+        for i in range(samples_number):
+            result = self.simulator.get_classical_simulation_results()
+
+            for j in range(len(possible_outcomes)):
+                # Check if obtained result is one of possible outcomes.
+                if not (result == possible_outcomes[j]).__contains__(False):
```

Comment:
Of course. It basically counts which sample has occured. If i-th sample has occured, then it's number count is increased. I cannot do that by all nor by any, because then I don't have the index. Is this explanation satisfactionary? I can provide a simple example if you'd like to see one.

Reply:
Have you considered the “False in container” notation? The only thing I frown upon here is the explicit use of the __contains__ method.

Reply:
I went with the `all` option,